calligraphy/src/main/java/uk/co/chrisjenx/calligraphy/CalligraphyUtils.java

```diff
@@ -346,7 +346,9 @@ static void applyExtraAttributesToTextView(final Context context, final TextView
                 }
             }
             appearance.recycle();
-            view.setLineSpacing(lineSpacingExtra, lineSpacingMultiplier);
+            if (view.getLineSpacingExtra() == 0 && view.getLineSpacingMultiplier() == 1) {
```

Comment:
I think you meant to check if this condition is NOT true.

Reply:
It's been a while, but I believe the intent here is that we'll only copy line spacing from the text style if the view itself doesn't have explicit overrides.

Reply:
Ahh... that makes sense. Thanks!

Reply:
So then wouldn't it make more sense to do this?

```
if (view.getLineSpacingExtra() == 0) {
  view.setLineSpacing(lineSpacingExtra, view.getLineSpacingMultiplier());
}
if (origLineSpacingMultiplier == 1) {
  view.setLineSpacing(view.getLineSpacingExtra(), lineSpacingMultiplier);
}
```

i.e. For cases where you specify a custom `android:lineSpacingExtra` value on the `TextView` and a custom `android:lineSpacingMultiplier` value on the text appearance?

Reply:
Actually maybe an even better option would be to check if `android:lineSpacingExtra` or `android:lineSpacingMultiplier` are being passed as attributes on the view... and ignore the values in the text appearance if they exist?

Reply:
oh, sure, perhaps. The diff was intended as a starting point & proof of concept, but given that it was rejected, and that the entire library is now essentially deprecated, it seems a moot discussion.

Reply:
Yeah... I'm actually using view pump to achieve this and was inspired by your idea... so thanks. :D

Reply:
what do you mean by "view pump" ?

Reply:
We built out into a more modular project: https://github.com/InflationX/ViewPump